### DISCUSSION TITLE
Since pydantic 1.10.3, SSLyze is broken

### DISCUSSION THREAD

**Discussion by KramNamez** [pydantic/pydantic#4905] [Question]
As seen in https://github.com/nabla-c0d3/sslyze/issues/586 SSLyze broke, despite pydantic only doing point releases that should not change its API.

Freezing pydantic at 1.10.2 fixes the issue for now, but I still wanted to ask if this is a bug in pydantic or in SSLyze's usage of pydantic.

**Comment by samuelcolvin** [comment]
This seems to be more of a complaint, than a bug report.

If there's a bug or breaking change in v1.10.3, please create an issue to report it, with as much detail as possible. 

If you just want to complain, please don't use open source, volunteer maintained software.

You might also include whether this is fixed in v1.10.4?

**Comment by KramNamez** [comment]
It's a request for information (also known as a question), because I don't know that it's pydantic that's broken, or SSLyze doing something wrong.

All I know is that if I `import sslyze` while I have pydantic 1.10.2 or lower installed, it works, and if I use 1.10.3 or 1.10.4, it doesn't, because `Cannot overwrite attribute __setattr__ in class CompressionScanResult`.

I can't really give you a detailed bug report, because that's it, and because I don't even know if pydantic is the right place to open a bug report. If you can point me at a way of figuring out where the bug is, I'll do that.

**Reply by samuelcolvin** [reply]
Thanks for adding more context.

Is this specific to python 3.7?

Most likely this was caused by #4880, which in turn was designed to fix other subtle problems with dataclasses. See #4880 added `__setattr__`. @PrettyWood might have some insights?

Honestly, I'm not sure where to go from here - dataclass support now has many many subtleties and ever fix we (mostly @PrettyWood :pray:) implements, seems to break some other edge case.

dataclass support should be radically simplified in V2, see #4670, but until then I'm not sure what we can really do about this.

**Comment by samuelcolvin** [comment]
Let's continue discussion on #4907, this should have been an issue, not a discussion.
